### PR TITLE
build: tests: add execution spec tests

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
This PR adds the execution spec tests to our test suite.
Doesn't work under windows yet

### CODE REVIEW

File: build/ci.go [hunk 1]
```diff
@@ -322,6 +325,29 @@ func doTest(cmdline []string) {
...
+	// Download the execution-spec-tests
+	const version = "0.2.4"
+
+	csdb := build.MustLoadChecksums("build/checksums.txt")
+	ext := ".tar.gz"
+	if runtime.GOOS == "windows" {
+		// TODO (MariusVanDerWijden) enable windows tests once spec tests provide a .zip
+		return ""
+		ext = ".zip"
+	}
```
Comment (holiman) [inline build/ci.go:338]: Why can't windows use the `.tar.gz` package?

Comment (MariusVanDerWijden) [reply build/ci.go:338]: I thought that windows can't do .tar.gz. but it looks like I'm wrong there. I haven't used windows in years

**Comment by holiman** [comment]
This PR
```
ok  	github.com/ethereum/go-ethereum/tests	314.167s
```

A random other PR
```
ok  	github.com/ethereum/go-ethereum/tests	322.125s
```
Are you sure it is actually running the tests? Or are there just not a lot of them?

File: build/ci.go [hunk 2]
```diff
@@ -322,6 +325,24 @@ func doTest(cmdline []string) {
...
 
+func downloadSpecTestFixtures(cachedir string) string {
+	// Download the execution-spec-tests
+	const version = "0.2.4"
+
+	csdb := build.MustLoadChecksums("build/checksums.txt")
+	ext := ".tar.gz"
+	base := "fixtures" // TODO(MariusVanDerWijden) rename once the version becomes part of the filename
+	url := fmt.Sprintf("https://github.com/ethereum/execution-spec-tests/releases/download/v%s/%s%s", version, base, ext)
+	archivePath := filepath.Join(cachedir, base+ext)
```
Comment (holiman) [inline build/ci.go:336]: If you take the `version` as a parameter, e.g. `go run ./build/ci.go -test --test.spec "0.2.4"`. Then we don't have to muck around with making version part of the filename. In order to update the spec, we just bump the version and update the hash. 

And then this PR would be good to go, without any TODO's left..(?)

Comment (holiman) [reply build/ci.go:336]: Actually, I guess it is already good to go. TO update for a new release of the tests, we have to update the `const version` string and the hash. Doing it the way I proposed doesn't simplify anything

File: build/ci.go [hunk 3]
```diff
@@ -284,12 +284,13 @@ func buildFlags(env build.Environment, staticLinking bool, buildTags []string) (
...
-		coverage = flag.Bool("coverage", false, "Whether to record code coverage")
-		verbose  = flag.Bool("v", false, "Whether to log verbosely")
-		race     = flag.Bool("race", false, "Execute the race detector")
+		dlgo         = flag.Bool("dlgo", false, "Download Go and build with it")
+		arch         = flag.String("arch", "", "Run tests for given architecture")
+		cc           = flag.String("cc", "", "Sets C compiler binary")
+		coverage     = flag.Bool("coverage", false, "Whether to record code coverage")
+		verbose      = flag.Bool("v", false, "Whether to log verbosely")
+		race         = flag.Bool("race", false, "Execute the race detector")
+		stateTestDir = "tests/spec-tests"
```
Comment (holiman) [inline build/ci.go:293]: Is there any reason for definining it here? If you move it into `downloadSpecTestFixtures`, you will have all related to spec-tests collected there, which might be nicer?

**Review by fjl — APPROVED** [review]

**Comment by fjl** [comment]
It would be nice to have a script/tool like `build/update-tests.go` that

- Downloads the latest tests release from the github repo
- Updates the checksum file
- Updates the version somewhere

**Pull request merged**